src-java/kilda-persistence-tinkerpop/src/test/java/org/openkilda/persistence/ferma/repositories/FermaHaFlowRepositoryTest.java

```diff
@@ -0,0 +1,301 @@
+        assertEquals(2, flowMap.size());
+
+        // flow 1
+        assertHaFlow(HA_FLOW_ID_1, SWITCH_ID_1, PORT_1, VLAN_1, INNER_VLAN_1, LATENCY_1, LATENCY_2, BANDWIDTH_1,
+                FlowEncapsulationType.TRANSIT_VLAN, PRIORITY_1, DESCRIPTION_1, PathComputationStrategy.COST,
+                FlowStatus.UP, true, true, true, true, true, flowMap.get(HA_FLOW_ID_1));
+        assertSubFlows(flowMap.get(HA_FLOW_ID_1).getSubFlows(), subFlow1, subFlow2);
+        assertPathsFlows(flowMap.get(HA_FLOW_ID_1).getPaths(), path1, path2);
+
+        // flow 2
+        assertHaFlow(HA_FLOW_ID_2, SWITCH_ID_2, PORT_2, VLAN_2, INNER_VLAN_2, LATENCY_3, LATENCY_4, BANDWIDTH_2,
+                FlowEncapsulationType.VXLAN, PRIORITY_2, DESCRIPTION_2, PathComputationStrategy.LATENCY,
+                FlowStatus.IN_PROGRESS, false, false, false, false, false, flowMap.get(HA_FLOW_ID_2));
+        assertEquals(0, flowMap.get(HA_FLOW_ID_2).getSubFlows().size());
+        assertEquals(0, flowMap.get(HA_FLOW_ID_2).getPaths().size());
+    }
+
+    @Test
+    public void existHaFlowTest() {
+        createHaFlow(haFlow1);
+        assertTrue(haFlowRepository.exists(haFlow1.getHaFlowId()));
+        assertFalse(haFlowRepository.exists(haFlow2.getHaFlowId()));
+    }
+
+    @Test
+    public void findByIdHaFlowTest() {
+        createHaFlow(haFlow1);
+        createHaFlow(haFlow2);
+        haSubFlowRepository.add(subFlow1);
+        haSubFlowRepository.add(subFlow2);
+        haFlow1.setSubFlows(Sets.newHashSet(subFlow1, subFlow2));
+
+        haFlowPathRepository.add(path1);
+        haFlowPathRepository.add(path2);
+        haFlow1.addPaths(path1, path2);
+
+        // flow 1
+        Optional<HaFlow> foundFlow1 = haFlowRepository.findById(HA_FLOW_ID_1);
+        assertTrue(foundFlow1.isPresent());
+        assertHaFlow(HA_FLOW_ID_1, SWITCH_ID_1, PORT_1, VLAN_1, INNER_VLAN_1, LATENCY_1, LATENCY_2, BANDWIDTH_1,
+                FlowEncapsulationType.TRANSIT_VLAN, PRIORITY_1, DESCRIPTION_1, PathComputationStrategy.COST,
+                FlowStatus.UP, true, true, true, true, true, foundFlow1.get());
+        assertSubFlows(foundFlow1.get().getSubFlows(), subFlow1, subFlow2);
+        assertPathsFlows(foundFlow1.get().getPaths(), path1, path2);
+
+        // flow 2
+        Optional<HaFlow> foundFlow2 = haFlowRepository.findById(HA_FLOW_ID_2);
+        assertTrue(foundFlow2.isPresent());
+        assertHaFlow(HA_FLOW_ID_2, SWITCH_ID_2, PORT_2, VLAN_2, INNER_VLAN_2, LATENCY_3, LATENCY_4, BANDWIDTH_2,
+                FlowEncapsulationType.VXLAN, PRIORITY_2, DESCRIPTION_2, PathComputationStrategy.LATENCY,
+                FlowStatus.IN_PROGRESS, false, false, false, false, false, foundFlow2.get());
+        assertEquals(0, foundFlow2.get().getSubFlows().size());
+        assertEquals(0, foundFlow2.get().getPaths().size());
+    }
+
+    @Test
+    public void findByIdNoExistentFlowTest() {
+        assertFalse(haFlowRepository.findById("123").isPresent());
```

Comment:
A bit tricky question what this test case proves. 
I suggest you combine it with the create or remove. You can assert that something was in the repo and it is not there after the removal. Or something was not in the repo and it is there after the creation. Or both.